crates/biome_html_parser/src/lexer/mod.rs

```diff
@@ -337,6 +319,37 @@ impl<'src> HtmlLexer<'src> {
 
         Ok(())
     }
+
+    /// Consume HTML text literals outside of tags.
+    ///
+    /// This includes text and single spaces between words. If newline or a second
+    /// consecutive space is found, this will stop consuming and to allow the lexer to
+    /// switch to `consume_whitespace`.
+    fn consume_html_text(&mut self) -> HtmlSyntaxKind {
```

Comment:
Is there a link to the spec that justify having newlines and double spaces parsed differently? It would be useful to have that link in the docs of the function

Reply:
I found this: https://infra.spec.whatwg.org/#strip-leading-and-trailing-ascii-whitespace

Reply:
My reasoning is that trivia means that the token doesn't affect the functionality of the code. The primary function of an HTML document is to render in a web browser. If you remove a space between words in HTML text, it does not render the document invalid, and it changes how the document is rendered (aka its functionality), and is therefore not trivia. Multiple spaces are collapsed into a single space, so all those extra spaces _are_ trivia.